### ISSUE TITLE
DSN not interpreted when using migration command (use Manager instead ?)

### ISSUE THREAD

**Issue by skwi** [cakephp/phinx#1406]
When using the command line to start a migration on an environment configured with DSN, I get the following error : 
```
Could not determine database name! Please specify a database name in your config file
```
It seems that the `Migrate` command relies on the instruction `$this->getConfig()->getEnvironment($environment);` to get the environment options which does not relies on the `Environment` class, where the `parseAgnosticDsn` can transform the DSN to a proper environment configuration.

The `$envOptions = $this->getManager()->getEnvironment($environment)->getOptions();` is able to fetch the proper configuration.

I can write the PR to use the manager in the command but before doing so I prefer checking with you that it's the proper way to do it and make sure I'm not missing something.

**Comment by lorenzo** [comment]
A PR would be great!

**Comment by djcvijic** [comment]
I also came across this error today.

@skwi I tried your approach, and while the DSN does get parsed now, tokens no longer get replaced! It's as if Config::replaceTokens() is never called, even though it seems like it should be called even before Manager is instantiated...

I have a hunch this is connected to https://github.com/cakephp/phinx/pull/1433 but I ran out of time to try to combine these two solutions for now. Will keep exploring at some point.

TL;DR do not implement this fix yet, it is an incomplete solution.

**Comment by alejandro-falkowski-gelato** [comment]
I see this is still an issue. Do you guys have an ETA when this will be released?
